server/src-lib/Hasura/Tracing.hs

```diff
@@ -197,42 +202,64 @@ instance MonadTrace m => MonadTrace (ExceptT e m) where
   currentReporter = lift currentReporter
   attachMetadata = lift . attachMetadata
 
+-- | Create a Random 16 character hex string
+randomHex :: IO BS.ByteString
```

Comment:
Is this a good way of creating a random hex value or should I use the Text.printf and convert the random Word64 using the `016x` ?

Reply:
Given we might want to support other propagation formats, I'd say it's fine to use a `ByteString` to store the 8 bytes, but we should store them unencoded in the context value, and encode them in a separate function which can be called in `injectHttpContext`. We might also want to create some newtype wrappers for trace and span IDs.

Reply:
I made a new encoding function and kept the ByteString type for the trace contexts. But I don't understand who do you mean by newtype wrappers for trace and span IDs and why do we need them. Could you please help me understand this.